text/0000-macros-1.1.md

```diff
@@ -0,0 +1,561 @@
+more principled `use` system in macros 2.0).
+
+### Initial implementation details
+
+This section lays out what the initial implementation details of macros 1.1
+will look like, but none of this will be specified as a stable interface to the
+compiler. These exact details are subject to change over time as the
+requirements of the compiler change, and even amongst platforms these details
+may be subtly different.
+
+The compiler will essentially consider `rustc-macro` crates as `--crate-type
+dylib -C prefer-dyanmic`. That is, compiled the same way they are today. This
+namely means that these macros  will dynamically link to the same standard
+library as the compiler itself, therefore sharing resources like a global
+allocator, etc.
+
+The `librustc_macro` crate will compiled as an rlib and a static copy of it
+will be included in each macro. This crate will provide a symbol known by the
+compiler that can be dynamically loaded. The compiler will `dlopen` a macro
+crate in the same way it does today, find this symbol in `librustc_macro`, and
+call it.
+
+The `rustc_macro_define` and `rustc_macro_derive` attributes will be encoded
+into the crate's metadata, and the compiler will discover all these functions,
+load their function pointers, and pass them to the `librustc_macro` entry point
+as well. This provides the opportunity to register all the various expansion
+mechanisms with the compiler.
+
+The actual underlying representation of `TokenStream` will be basically the same
+as it is in the compiler today. (the details on this are a little light
+intentionally, shouldn't be much need to go into *too* much detail).
+
+### Initial Cargo integration
+
+Like plugins today, Cargo needs to understand which crates are `rustc-macro`
+crates and which aren't. Cargo additionally needs to understand this to sequence
+compilations correctly and ensure that `rustc-macro` crates are compiled for the
+host platform. To this end, Cargo will understand a new attribute in the `[lib]`
+section:
+
+```toml
+[lib]
+rustc-macro = true
+```
+
+This annotation indicates that the crate being compiled should be compiled as a
+`rustc-macro` crate type for the host platform in the current compilation.
+
+Eventually Cargo may also grow support to understand that a `rustc-macro` crate
+should be compiled twice, once for the host and once for the target, but this is
+intended to be a backwards-compatible extension to Cargo.
+
+## Pieces to stabilize
+
+Eventually this RFC is intended to be considered for stabilization (after it's
+implemented and proven out on nightly, of course). The summary of pieces that
+would become stable are:
+
+* The `rustc_macro` crate, and a small set of APIs within (skeleton above)
+* The `rustc-macro` crate type, in addition to its current limitations
+* The `#[rustc_macro_derive]` attribute
+* The signature of the `#![rustc_macro_derive]` functions
+* The `#![rustc_macro_crate]` attribute
```

Comment:
Oops, thanks for the sharp eye! These were actually remnants of a previous draft, I just forgot to remove them.